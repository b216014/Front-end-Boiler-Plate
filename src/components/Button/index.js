import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const Button = (props) => {
  const { testId, category, onClick } = props;
  return (
    <div data-testid={testId}>
      <button type="button" className={styles.button} onClick={onClick}>
        {category}
      </button>
    </div>

  );
};

Button.propTypes = {
  testId: propTypes.string.isRequired,
  category: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
};

export default Button;
