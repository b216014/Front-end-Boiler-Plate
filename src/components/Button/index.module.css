.button{
    display: flex;
    height: 3rem;
    /* justify-content: center; */
    width: 9rem;
    justify-content: center;
    align-items: center;
    border: 1px solid black;
    border-radius: 21%;
  }

.card{
  display: flex;
  flex-direction: column;
}