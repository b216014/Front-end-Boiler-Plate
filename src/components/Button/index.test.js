import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';


describe('the button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button testID="test-btn">Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testId="test-btn" click={onClickMock} />);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });

  it('should verify if the count value is right or not', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testId="test-btn" buttonType="rect" click={onClickMock} category="All" />);
    fireEvent.click(getByTestId('test-btn'));
    // expect(getByTestId('test-btn')).toHaveTextContent(category);
  });
});
