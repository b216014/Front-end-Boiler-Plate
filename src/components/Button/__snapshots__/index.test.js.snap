// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`the button component should verify the type of button passed (round) to the component 1`] = `undefined`;
